### PR TITLE
[Console] Allow a SymfonyStyle instance to return option user input

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
| Q | A |
| --- | --- |
| Branch? | "master" |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| License | MIT |

As explained here: http://stackoverflow.com/questions/40150379/how-to-use-a-symfonystyle-object-to-get-options

With this commit it will be truly possible to use an object `$io` instance of `SymfonyStyle` class to be used as both input/output.

### CODE REVIEW

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 1]
```diff
@@ -425,4 +425,14 @@ private function createBlock($messages, $type = null, $style = null, $prefix = '
 
         return $lines;
     }
+
+    /**
+     * @param string $name
+     *
+     * @return mixed
+     */
+    public function getOption($name)
```
Comment (stof) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: public methods must go before private ones

**Review by stof — COMMENTED** [review]
Having `getOption` without `getArgument` does not make sense IMO

File: src/Symfony/Component/Console/Style/SymfonyStyle.php (hunk 1 above)
Comment (ShinDarth) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: @stof updated, thanks

**Comment by chalasr** [comment]
👍  For accessing to some part of the Input api, as we did for the `Output` in `OutputStyle` (verbosity and decoration).

As said previously, `getArgument()` should be added as well.
In the same spirit, what about adding `hasOption()`/`hasArgument()` and `getOptions()`/`getArguments()` too?

**Comment by ro0NL** [comment]
So.. do we need a IO interface then? As right now.. this is *not* a IO object;

`SymfonyStyle extends OutputStyle implements OutputInterface, StyleInterface`

I suggest implementing `InputInterface` somewhere down the chain.

**Comment by ogizanagi** [comment]
I'm not convinced the styles should act as inputs. Styles are helpers meant to **standardize the console output** across the application. Methods requiring inputs like `ask()` only does because asking a question to the user also requires outputting something, and thus, standardization. 
The pseudo decoration of the `OutputInterface` makes sense in this way, but IMHO it doesn't for `InputInterface` regarding the styles responsibility.

**Comment by chalasr** [comment]
@ro0NL I agree it should be clearer that the implementation gives an IO object, as we are naming it and using it as well. However I don't see any benefit in making `SymfonyStyle` able to alter the input definition in addition of accessing it, thus not sure it should implement `InputInterface`

**Comment by ogizanagi** [comment]
We should keep in mind `$io` is just an handy convention to name the style helper in your code. It does not reflect well the object responsibility if you compare it to both `Output/InputInterface`, but is convenient.

**Comment by ro0NL** [comment]
Imo. having a technically correct IO object (implement both interfaces) is eventually most convenient, as you can use it in many cases, and apply utilization that depends on IO traffic.

Agree `Style` implies a helper object, not an IO. Again, this doesn't mean having an IO is not useful as do think a helper object (in this case the style helper) _should_ depend on the IO object.

Meaning;

```php
$io = new IO($input, $output);
$style = new SymfonyStyle($io);

// when passing $style around
$io = $style->getIO(); // useful ;-)
```

**Comment by ogizanagi** [comment]
@ro0NL : Agreed. It should be an entirely new object (if really necessary in the Symfony codebase...)

**Comment by ro0NL** [comment]
I think it's totally worth it.. before `$io = new Style()` gets out of hand. Hence this kind of PR.

**Comment by ogizanagi** [comment]
Beyond the current misunderstanding regarding the style variable naming, what would exactly be the benefits ?

**Comment by ro0NL** [comment]
> apply utilization that depends on IO traffic

Ie. being able to explicitly depend on `__construct(IOInterface $io)` vs. `__construct(InputInterface $input, OutputInterface $output)`.

This is not a must-have, but a nice-to-have :)

**Comment by ShinDarth** [comment]
Dear all,

if we all agree on a specific solution for this problem, I would like to work on it.

Please let me know.

**Comment by ShinDarth** [comment]
@stof I'm tagging you as (afaik) the only merger for this component. Maybe you can tell me what can be a good solution to work on in order to solve this.

**Pull request closed without merging**